replica/table.cc

```diff
@@ -582,17 +589,19 @@ storage_group& storage_group_manager::storage_group_for_id(const schema_ptr& s,
 class single_storage_group_manager final : public storage_group_manager {
     replica::table& _t;
     storage_group* _single_sg;
+    compaction_group* _single_cg;
 
     compaction_group& get_compaction_group() const noexcept {
-        return const_cast<compaction_group&>(*_compaction_groups.begin());
+        return *_single_cg;
```

Comment:
Why not `*_single_sg->main_compaction_group().get()`?

Reply:
sure, why not. even though this is a hot path, compiler should be able to optimize it easily